blake2/src/impl_hacl/blake2b.rs

```diff
@@ -0,0 +1,145 @@
+use libcrux_hacl_rs::streaming_types::error_code;
+
+use crate::hacl::hash_blake2b::{
+    blake2_params, digest, index, malloc_raw, params_and_key, reset, reset_with_key, state_t,
+    update0,
+};
+
+use crate::impl_hacl::Error;
+
+const PARAM_LEN: usize = 16;
+const MAX_LEN: usize = 64;
+
+/// A builder for [`Blake2b`]. `KEY_LEN`` must be in the range `0..=64``, `OUT_LEN`` must be in the
+/// range `1..=64`.
+pub struct Blake2bBuilder<'a, const KEY_LEN: usize, const OUT_LEN: usize> {
+    key: &'a [u8; KEY_LEN],
+    personal: &'a [u8; PARAM_LEN],
+    salt: &'a [u8; PARAM_LEN],
+}
+
+impl<const OUT_LEN: usize> Blake2bBuilder<'_, 0, OUT_LEN> {
+    /// Creates the builder for an unkeyed hasher and returns an error if the digest length
+    /// `OUT_LEN` is not in the allowed range.
+    pub fn new_unkeyed() -> Result<Self, Error> {
+        if OUT_LEN < 1 || OUT_LEN > MAX_LEN {
+            return Err(Error::InvalidDigestLength);
+        }
+
+        Ok(Self {
+            key: &[],
+            personal: &[0; PARAM_LEN],
+            salt: &[0; PARAM_LEN],
+        })
+    }
+}
+impl<'a, const KEY_LEN: usize, const OUT_LEN: usize> Blake2bBuilder<'a, KEY_LEN, OUT_LEN> {
+    /// Creates the builder and returns an error if the lengths `KEY_LEN` or `OUT_LEN` are not in
+    /// the allowed range.
+    pub fn new() -> Result<Self, Error> {
```

Comment:
I'm not sure if I understand how this can be used without a length that's known at compile time.
How would I implement the following function with this implementation?
```rust
fn blake2b(out: &mut [u8]) {
    // Call libcrux blake2b to write out.len() bytes
}
```

Reply:
Oh, I just realized I didn't communicate very clearly. So far I only did Blake2s and I wanted to check with you before applying the same pattern to Blake2b.